### PR TITLE
fix: backport #1949 + fix handler entry

### PR DESCRIPTION
**Pull request by bermuell**
## Description

- backport #1949
- fix: added missing handler entry for 'consumer modification proposal'

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [x] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

## Summary by CodeRabbit

- **New Features**
  - Introduced `ConsumerModificationProposal` for dynamically altering power shaping parameters and switching between Top N and Opt-In consumer chains.

- **Documentation**
  - Added tips for transitioning consumer chains and modifying power shaping parameters.
  - Updated URLs for validator setup and Cosmos SDK documentation.
  - Clarified proposal types in the Interchain security module.

- **Bug Fixes**
  - Corrected a typo in function comments.

- **Tests**
  - Added E2E tests for `ConsumerModificationProposal` and related security parameter modifications.

### CODE REVIEW

File: .github/workflows/nightly-e2e.yml [hunk 1]
```diff
@@ -277,6 +277,22 @@ jobs:
...
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+      - uses: actions/checkout@v4
+      - name: Checkout LFS objects
+        run: git lfs checkout
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.21" # The Go version to download (if necessary) and use.
```
Comment (sainoe) [inline .github/workflows/nightly-e2e.yml:293]:
```suggestion
          go-version: "1.22" # The Go version to download (if necessary) and use.
```

File: docs/docs/validators/joining-testnet.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The experience gained in the testnet will prepare you for validating interchain
 :::tip
 Provider and consumer chain represent distinct networks and infrastructures operated by the same validator set.
 
-For general information about running cosmos-sdk based chains check out the [validator basics](https://hub.cosmos.network/validators/validator-setup) and [Running a Node section](https://docs.cosmos.network/main/run-node/run-node) of Cosmos SDK docs
+For general information about running cosmos-sdk based chains check out the [validator basics](https://hub.cosmos.network/main/validators/validator-setup) and [Running a Node section](https://docs.cosmos.network/main/run-node/run-node) of Cosmos SDK docs
```
Comment (sainoe) [inline docs/docs/validators/joining-testnet.md:15]:
```suggestion
For general information about running cosmos-sdk based chains check out the [validator basics](https://hub.cosmos.network/main/validators/validator-setup) and [Running a Node section](https://docs.cosmos.network/main/run-node/run-node) of Cosmos SDK docs.
```

**Review by sainoe — APPROVED** [review]
LGTM

File: .github/workflows/nightly-e2e.yml (hunk 1 above)
Comment (MSalopek) [reply .github/workflows/nightly-e2e.yml:293]: Please bump it to v1.22

**Review by MSalopek — APPROVED** [review]

**Pull request merged**